racetrack_module.py

```diff
@@ -0,0 +1,588 @@
+import math
+import os
+from typing import Union, Tuple, List, Dict
+from dataclasses import dataclass
+
+from PyQt5.QtWidgets import (
+    QSpinBox, QComboBox, QFileDialog, QDialog,
+    QVBoxLayout, QHBoxLayout, QLabel, QPushButton,
+)
+from qgis._core import (
+    QgsWkbTypes, Qgis,
+    QgsGeometry, QgsVector, QgsPointXY,
+    QgsFeature, QgsExpressionContextUtils,
+    QgsVectorLayer,
+    QgsFields, QgsField, QgsCoordinateReferenceSystem, QgsCoordinateTransform, QgsProject, QgsUnitTypes,
+)
+from qgis.gui import QgisInterface
+from PyQt5.QtCore import QVariant
+from .coverage_module import CoverageModule
+from .utils import LayerUtils
+
+DEFAULT_MAX_TURN_DISTANCE = 1000
+
+@dataclass
+class FlightParameters:
+    flight_altitude: float
+    overlap: float
+    coverage_range: float
+    sensor_name: str
+    max_turn_distance: float
+    algorithm: str
+
+
+class RacetrackDialog(QDialog):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        # Get the saved max turn distance or use default value 1000
+        self.max_turn_distance = QgsProject.instance().readDoubleEntry(
+            "ScienceFlightPlanner", "max_turn_distance", DEFAULT_MAX_TURN_DISTANCE
+        )[0]
```

Comment:
Maybe replace with
```suggestion
        self.max_turn_distance, _= QgsProject.instance().readDoubleEntry(
            "ScienceFlightPlanner", "max_turn_distance", DEFAULT_MAX_TURN_DISTANCE
        )
```
for better clarity